src/model-viewer.ts

```diff
@@ -21,27 +21,18 @@ import {LoadingMixin} from './features/loading.js';
 import {MagicLeapMixin} from './features/magic-leap.js';
 import {StagingMixin} from './features/staging.js';
 import ModelViewerElementBase from './model-viewer-base.js';
-import {Constructor} from './utilities.js';
 import {FocusVisiblePolyfillMixin} from './utilities/focus-visible.js';
 
-type ModelViewerMixin =
-    (ModelViewerElement: Constructor<ModelViewerElementBase>) =>
-        Constructor<ModelViewerElementBase>;
+export const ModelViewerElement = MagicLeapMixin(
+    StagingMixin(EnvironmentMixin(ControlsMixin(ARMixin(LoadingMixin(
+        AnimationMixin(FocusVisiblePolyfillMixin(ModelViewerElementBase))))))));
 
-const mixins = [
-  FocusVisiblePolyfillMixin,
-  AnimationMixin,
-  LoadingMixin,
-  ARMixin,
-  ControlsMixin,
-  EnvironmentMixin,
-  StagingMixin,
-  MagicLeapMixin
-];
+export type ModelViewerElement = InstanceType<typeof ModelViewerElement>;
```

Comment:
@rictic @aomarks could you comment on whether or not this is a reasonable approach to ensuring this element implementation is properly typed, given our usage of class mixins?

Reply:
Looks good to me! I hadn't seen the `InstanceType<typeof MixinApplication>` pattern before -- that's really great! It solves the problem where the result of a mixin application call is a value but not a type in a really simple way.